### PR TITLE
234 create api endpoint for single location

### PR DESCRIPTION
**Pull request by Lucas-Nan**
LoadLocationData endpoint - may currently be to slow for actual use.
Refactored other endpoints a bit with id as parameter

### CODE REVIEW

File: backend/src/BIE.Core/BIE.Core.API/Controllers/DatasetController.cs [hunk 1]
```diff
@@ -143,7 +160,81 @@ FROM dbo.Hausumringe_mittelfranken
...
+                var response = new LocationDataResponse
+                {
+                    CurrentDatasetData = new List<DatasetItem>()
+                };
+
+                foreach (var row in DbHelper.GetData(formattedQuery))
+                {
+                    var area = row["Area"];
+                    var distance = row["Distance"];
+                    var location = row["Location"];
```
Comment (Ahsankkhan) [inline backend/src/BIE.Core/BIE.Core.API/Controllers/DatasetController.cs:216]: The query is not returning any location parameter. That why getting exception over here.

Comment (Ahsankkhan) [inline backend/src/BIE.Core/BIE.Core.API/Controllers/DatasetController.cs:216]: The query is not returning any location parameter. That why getting exception over here.

Comment (Lucas-Nan) [reply backend/src/BIE.Core/BIE.Core.API/Controllers/DatasetController.cs:216]: oh i didnt even get this exception^^ i already got a timeout before that :D

Comment (Lucas-Nan) [reply backend/src/BIE.Core/BIE.Core.API/Controllers/DatasetController.cs:217]: so i added location but cant test it on my side. Can you check as well @Ahsankkhan ?

**Review by Ahsankkhan — DISMISSED** [review]
Its working after adding the timeout. Currently taking more time in execution.

**Review by Ahsankkhan — APPROVED** [review]

**Pull request merged**